.github/workflows/integration.yml

```diff
@@ -54,6 +54,19 @@ jobs:
            echo "::set-output name=repo-name::${{ github.event.client_payload.github.payload.repository.full_name }}"
         fi
 
+      # Since we are using a repository-dispatch event, we have to explicitly set a run check. We initialize it to a "pending" state.
+    - uses: octokit/request-action@v2.x
+      id: create_check_run
+      name: "Initialize run check to 'pending (For PR-only)"
+      with:
+        route: POST /repos/${{ github.repository }}/statuses/${{ steps.configure.outputs.commit-ref }}
+        state: "pending"
+        description: "Component build status"
+        context:  "Components building"
+        target_url: "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+      env:
+        GITHUB_TOKEN: ${{ secrets.CI_TOKEN }}
```

Comment:
Should be probably executed only if the event is repository-dispatch. Same for the other occurrences.